modules/dataframe/src/jsondataframeconversion.cpp

```diff
@@ -0,0 +1,154 @@
+/*********************************************************************************
+ *
+ * Inviwo - Interactive Visualization Workshop
+ *
+ * Copyright (c) 2019 Inviwo Foundation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *********************************************************************************/
+#pragma once
+
+#include <inviwo/dataframe/jsondataframeconversion.h>
+#include <inviwo/core/datastructures/buffer/bufferramprecision.h>
+#include <iterator>
+#include <vector>
+
+namespace inviwo {
+
+namespace plot {
+
+namespace detail {
+
+// Helper for adding columns based on json item type
+void addDataFrameColumnHelper(json::value_t valueType, std::string header, DataFrame& df) {
+    switch (valueType) {
+        case json::value_t::null:  ///< null value
+            // need to check more rows to know type
+            break;
+        case json::value_t::object:  ///< object (unordered set of name/value pairs)
+            // Not supported
+            break;
+        case json::value_t::array:  ///< array (ordered collection of values)
+            // Not supported
+            break;
+        case json::value_t::string:  ///< string value
+            df.addCategoricalColumn(header, 0u);
+            break;
+        case json::value_t::boolean:  ///< boolean value
+            // FIXME: Get compile error when trying to add bool column
+            // Error C2039 'data' : is not a member of 'std::vector<bool,std::allocator<_Ty>>' 
```

Comment:
This is not an error really, vector<bool> does not have a data menber function... have a look at vector<bool> to see why..